### PR TITLE
docs: improve installation documentation

### PR DESCRIPTION
**Pull request by anuraghazra**
## Component Checklist

- [ ] Update Component Status Page
- [ ] Perform Manual Testing in Other Browsers
- [ ] Add KitchenSink Story
- [ ] Add Interaction Tests (if applicable)
- [ ] Add changeset

### CODE REVIEW

File: packages/blade/docs/guides/Installation.stories.mdx [hunk 1]
```diff
@@ -1,182 +1,290 @@
...
+    </TabList>
+  </Box>
+  <TabPanel value="basic">
+    <Heading size="large" marginTop="spacing.7">Add blade to your application</Heading>
+
+    <StepGroup orientation="vertical" size="medium">
+      <StepItem
+        title="1. Install blade as a dependency."
+      >
+        <Text>Blade has a peer dependency on a few libraries, you can skip adding it if you already have it installed in your project.</Text>
```
Comment (snitin315) [inline packages/blade/docs/guides/Installation.stories.mdx:90]:
```suggestion
        <Text>Blade has a few peer dependencies that you may already have installed in your project. If so, you can skip adding them again.</Text>
```

File: packages/blade/docs/guides/Installation.stories.mdx [hunk 2]
```diff
@@ -1,182 +1,290 @@
...
+            </ListItemText>
+          </ListItem>
+        </List>
+
+      </StepItem>
+      <StepItem
+        title="2. Add ReactNative Dependencies [Optional for web projects]"
+      >
+        ```shell
+        yarn add @floating-ui/react-native@0.10.0 react-native-reanimated@3.4.1 react-native-tab-view@3.5.2 react-native-pager-view@6.2.1 react-native-svg@12.3.0 react-native-gesture-handler@2.9.0 @gorhom/bottom-sheet@4.4.6 @gorhom/portal@1.0.1
```
Comment (snitin315) [inline packages/blade/docs/guides/Installation.stories.mdx:133]: This is making the X-axis overflow, 

https://github.com/razorpay/blade/assets/46647141/097f9c3a-9ce0-4801-860d-f1fe6c80fecf

File: packages/blade/docs/guides/Installation.stories.mdx [hunk 3]
```diff
@@ -1,182 +1,290 @@
...
+      <TabItem value="fonts">2. Install Fonts</TabItem>
+      <TabItem value="ts">3. TypeScript</TabItem>
+    </TabList>
+  </Box>
+  <TabPanel value="basic">
+    <Heading size="large" marginTop="spacing.7">Add blade to your application</Heading>
+
+    <StepGroup orientation="vertical" size="medium">
+      <StepItem
+        title="1. Install blade as a dependency."
```
Comment (snitin315) [inline packages/blade/docs/guides/Installation.stories.mdx:88]:
```suggestion
        title="1. Install @razorpay/blade as a dependency."
```

File: packages/blade/docs/guides/Installation.stories.mdx [hunk 4]
```diff
@@ -1,182 +1,290 @@
...
+
+  </TabPanel>
+  <TabPanel value="fonts">
+    <Heading size="large" marginTop="spacing.7">Installing Fonts</Heading>
+
+    We use 2 fonts. [TASA Orbiter](https://tasatype.localremote.co/) (for our headings), and [Inter](https://rsms.me/inter/) (for other text elements).
+
+    ### Web
+
+    - To add these fonts to your project, you can import them from blade package in your root entry file.
```
Comment (snitin315) [inline packages/blade/docs/guides/Installation.stories.mdx:170]:
```suggestion
    - To add these fonts to your project, import them from the blade package in your root entry file.
```

File: packages/blade/docs/guides/Installation.stories.mdx [hunk 5]
```diff
@@ -1,182 +1,290 @@
...
+
+      ```ts
+      // Somewhere in root index.ts or entryBrowser.tsx file
+      import '@razorpay/blade/fonts.css';
+      ```
+
+    - While Blade handles setting font-family for its own components, you can set font-family globally as well in your global styles for any custom components or exceptions
```
Comment (snitin315) [inline packages/blade/docs/guides/Installation.stories.mdx:177]:
```suggestion
    - While Blade handles the `font-family` for its components, you can set the `font-family` globally in your project's global styles for any custom components or exceptions.
```

File: packages/blade/docs/guides/Installation.stories.mdx [hunk 6]
```diff
@@ -1,182 +1,290 @@
...
+
+      h1, h2, h3, h4, h5, h6 {
+        font-family: ${(props) => props.theme.typography.fonts.family.heading};
+      }
+      `;
+      ```
+
+    ##### No Bundler Setups or No Code Tools
+
+    If your project uses a no-bundler setup or is a no-code tool, you can install fonts with `unpkg` link
```
Comment (snitin315) [inline packages/blade/docs/guides/Installation.stories.mdx:201]:
```suggestion
    If your project uses a no-bundler setup or is a no-code tool, you can install fonts with the `unpkg` link
```

File: packages/blade/docs/guides/Installation.stories.mdx [hunk 7]
```diff
@@ -1,182 +1,290 @@
...
-  // Somewhere in root index.ts or entryBrowser.tsx file
-  import '@razorpay/blade/fonts.css';
-  ```
+        ```js
+        // add the below import statement after all the import statements
+        import com.facebook.react.views.text.ReactFontManager;
+        ```
 
-- While Blade handles setting font-family for its own components, you can set font-family globally as well in your global styles for any custom components or exceptions
+      - Now search for `onCreate` method and add the following line to it
```
Comment (snitin315) [inline packages/blade/docs/guides/Installation.stories.mdx:243]:
```suggestion
      - Now search for the`onCreate` method and add the following line to it
```

File: packages/blade/docs/guides/Installation.stories.mdx [hunk 8]
```diff
@@ -1,182 +1,290 @@
...
-    ```js
-    module.exports = {
-      // ... rest of the config
-      assets: ['./public/fonts/'],
-    };
-    ```
+    ### Web
 
-  - Run the following command to link fonts in ios
+    If you're using TypeScript in your project, firstly make sure you have `@types/styled-components` installed in your project.
```
Comment (snitin315) [inline packages/blade/docs/guides/Installation.stories.mdx:260]:
```suggestion
    If you're using TypeScript in your project, first make sure you have `@types/styled-components` installed.
```

File: packages/blade/docs/guides/Installation.stories.mdx [hunk 9]
```diff
@@ -1,182 +1,290 @@
...
 
     ```shell
-    npx react-native-asset -ios-a
+    yarn add -D @types/styled-components
     ```
 
-- #### For Android
+    Now, you can extend the theme interface to include your custom theme properties to styled-components.
```
Comment (snitin315) [inline packages/blade/docs/guides/Installation.stories.mdx:266]:
```suggestion
    Now, you can extend the theme interface to include your custom theme properties to `styled-components`.
```

File: packages/blade/docs/guides/Installation.stories.mdx [hunk 10]
```diff
@@ -1,182 +1,290 @@
...
-    import com.facebook.react.views.text.ReactFontManager;
+    declare module 'styled-components' {
+      export interface DefaultTheme extends Theme {}
+    }
     ```
 
-  - Now search for `onCreate` method and add the following line to it
+    ### React Native
+
+    To get ReactNative specific type hints for blade you can switch the `moduleSuffixes` in your `tsconfig.json` to include `.native` files.
```
Comment (snitin315) [inline packages/blade/docs/guides/Installation.stories.mdx:279]:
```suggestion
    To get ReactNative specific type hints for the blade components, you can switch the `moduleSuffixes` in your `tsconfig.json` to include `.native` files.
```

File: packages/blade/docs/guides/Installation.stories.mdx (hunk 2 above)
Comment (anuraghazra) [reply packages/blade/docs/guides/Installation.stories.mdx:133]: Can't do much, storybook forces the codeblock to be fixed width.

Comment (saurabhdaware) [reply packages/blade/docs/guides/Installation.stories.mdx:133]: can you add Box wrapper around that codeblock with fixed width and overflow auto?

**Review by saurabhdaware — COMMENTED** [review]
Nice enhancements 🥳 

Just this comment - https://github.com/razorpay/blade/pull/2286#discussion_r1669744719

Rest looks good

**Review by saurabhdaware — APPROVED** [review]

**Pull request merged**